### ISSUE TITLE
[V2] iOS back gesture doesn't work when topBar or backButton is hidden

### ISSUE THREAD

**Issue by wsliaw** [wix/react-native-navigation#4455]
### Issue Description

Back gesture doesn't work when topBar or backButton's visible set to false

### Steps to Reproduce / Code Snippets / Screenshots

```javascript
Navigation.push(this.props.componentId, {
  component: {
    name: 'example.PushedScreen',
    passProps: {
      text: 'Pushed screen'
    },
    options: {
      topBar: {
        visible: false
      }
    }
  }
});
```
or
```javascript
Navigation.push(this.props.componentId, {
  component: {
    name: 'example.PushedScreen',
    passProps: {
      text: 'Pushed screen'
    },
    options: {
      topBar: {
        backButton: {
          visible: false
        }
      }
    }
  }
});
```

---
### Environment
* React Native Navigation version: 2.2.1-snapshot.61 
* React Native version: 0.57.7
* Platform(s): iOS
* Device info: Both simulator & device, iOS 12, Both debug & release

**Comment by ptfly** [comment]
Doesn't matter (in my case) if it's hidden or not. As soon as you're using custom left button, the gesture is gone! Was using 2.1.3 and it was working fine (now on 2.2.5).

**Comment by ptfly** [comment]
It's still happening on 2.4.0, any idea how to fix it?

**Comment by RichardLindhout** [comment]
@ptfly I still have the same problem on 2.5.0.

**Comment by RichardLindhout** [comment]
Probably a very dirty fix. Since I don't have any experience in iOS.

I did some testing with answers from the stackoverflow page @SasjaVDD suggested.

This works for me:
1. Open xCode
2. Open Libraries>ReactNativeNavigation.xcodeproj
3. Open file: UINavigationController+RNNOptions.m
4. Replace
```
- (void)rnn_setNavigationBarVisible:(BOOL)visible animated:(BOOL)animated {
	[self setNavigationBarHidden:!visible animated:animated];
}
```
5. With this
```
- (void)rnn_setNavigationBarVisible:(BOOL)visible animated:(BOOL)animated {
	[self setNavigationBarHidden:!visible animated:animated];
	
	// enable slide-back
	if ([self.navigationController respondsToSelector:@selector(interactivePopGestureRecognizer)]) {
		self.interactivePopGestureRecognizer.enabled = YES;
		self.interactivePopGestureRecognizer.delegate = nil;
	}
}
```

I'm setting a delegate to nil. I don't know what the unintended consequences are.

**Comment by Rigellute** [comment]
Also having this issue

**Comment by pie6k** [comment]
It's not stale - I'm still having this issue

**Comment by oblador** [comment]
This is still relevant :-) There's a PR up fixing the problem since two months, I think it deserves a review.

**Comment by pie6k** [comment]
Is this released already?

**Comment by vincentzierigen** [comment]
Is this fixed? I'm on 3.0.0-alpha.10 and it seems not to work
